### ISSUE TITLE
SEPA debit payment not working with Woopayments

### ISSUE THREAD

**Issue by Lockoloop** [Automattic/woocommerce-payments#8476]
### Describe the bug

> 💡 Note: this issue was further refined and the Acceptance Criteria for it are summarized in https://github.com/Automattic/woocommerce-payments/issues/8476#issuecomment-2078056929.

When my customer tries to pay via SEPA direct debit, their order fails.

### To Reproduce
1. Customer places an order.
2. Tries to pay via SEPA direct debit
3. The order shows as failed

### Actual behavior
The SEPA Direct Debit method of payment is enabled on the WooPayments page. 
This error message shows in the order notes:
> The PaymentMethod provided (sepa_debit) is not allowed for this PaymentIntent. Please attach a PaymentMethod of one of the following types: card. Alternatively, update the allowed payment_method_types for this PaymentIntent to include "sepa_debit".

### Expected behavior
The customer should be able to complete their transaction.

### Desktop (please complete the following information):
* Plugin Version 7.3.0

### Additional context
This customer has successfully used SEPA direct debit in the past, although I'm unsure if I were also using Stripe payments plugin on top of WooPayments back then.

**Comment by timur27** [comment]
Hey @Lockoloop, thanks for the report. I've been trying to reproduce it but with no success so far, as I'm able to successfully place an order paying with SEPA on my test site with WooPayments `7.3.0`. 

![image](https://github.com/Automattic/woocommerce-payments/assets/22319444/719e7cd0-b28e-48b6-84d9-47023a6c0fe2)


> although I'm unsure if I were also using Stripe payments plugin on top of WooPayments back then.

Could we try to turn off Stripe payments plugin for testing purposes and confirm if the failing SEPA if indeed coming from WooPayments? Thank you.

**Comment by csmcneill** [comment]
This could be related to https://github.com/Automattic/woocommerce-payments/issues/3776

**Comment by Lockoloop** [comment]
Hi Timur,
It was already off; I've removed the Stripe plugin some time ago already, when I stopped having issues with WooPayments (#7482) but only after that customer last ordered.
I haven't gotten any SEPA payments since I turned off Stripe.

**Comment by timur27** [comment]
@csmcneill please correct me if I'm wrong but it seems that https://github.com/Automattic/woocommerce-payments/issues/3776 covers subscription renewals while this issue covers placing an order?

**Comment by csmcneill** [comment]
@timur27 That's correct — but it's the only other instance I can find of this specific error.

**Comment by FangedParakeet** [comment]
Just want to add that I was also unable to replicate the issue described here--using WooPayments v.7.3.0 both locally and on a Jurassic Ninja test site. @csmcneill & @Lockoloop, we may need some more details about the specifics of the store in question here to aid us in replicating and resolving the scenario described herein. 🙏

**Comment by csmcneill** [comment]
Thanks for looking into this @FangedParakeet!

@Lockoloop Since we need some specific details about your site and how it is set up, we would prefer not to handle further troubleshooting on a public forum. Could you please navigate to [Woo.com > My Account > Support](https://woo.com/my-account/contact-support/) and open up a chat or ticket with us so that we can look into this further? You may need to create an account before you can access that page.

Please include a link to this GitHub issue so that our Happiness Engineers can have a more complete picture of what's going on :)

**Comment by Lockoloop** [comment]
Hi, sorry for the late reply but I had to check with them what was happening.
What happened is that the bank account used for SEPA Payment is being saved in the Credit Card frame as well, and since both look similar, my customer clicked on payment by card. The SEPA payment works. So the issue is that the account number is saved as a credit card number.
![credit_card_bug](https://github.com/Automattic/woocommerce-payments/assets/94473878/eb75d286-b3e8-43c4-978d-c987289b30c9)

**Comment by Lockoloop** [comment]
Hi @timur27 , have you been able to check my last comment?

**Comment by csmcneill** [comment]
Howdy @Lockoloop! I worked with Timur a bit on this today, and we've been able to reproduce the issue you specified in https://github.com/Automattic/woocommerce-payments/issues/8476#issuecomment-2045092387.

### Steps to Reproduce
1. Enable SEPA for your WooPayments account.
2. Revert to version 6.4.0 or earlier (I tested with version 6.3.0).
3. Make a payment using SEPA and choose to save your payment method.
4. Update WooPayments to version 7.5.3.
5. Attempt to purchase another product.
6. Use the saved SEPA payment method that appears under the `Credit card / debit card` radio option.
7. Note the error message on the checkout page.
8. Navigate to **WooCommerce > Orders**.
9. Locate the `Failed` order associated with the attempted purchase made in steps 5–6.
10. Assess the order notes.

### Background and Screenshots
Currently, customers add a new saved SEPA payment method, but they were able to up through version 6.4.0. This was also when we were using a different implementation of the Universal Payment Element (or `UPE`) which included card payments + SEPA payments within the same element:

![image](https://github.com/Automattic/woocommerce-payments/assets/82999806/1f84175b-d22d-4d85-a892-3f46c37f71cb)

Since then, we split payment methods into individual radio buttons on checkout (which we call `Split UPE`, since payment methods are split up), but saved SEPA payment methods are still rendered in the same spot as they were in 6.4.0:

![image](https://github.com/Automattic/woocommerce-payments/assets/82999806/56349d85-7bf6-45f0-a066-1440c7aeab63)

After this change, we are seeing that saved SEPA payment methods are also rendered under the `SEPA Direct Debit` radio option:

![image](https://github.com/Automattic/woocommerce-payments/assets/82999806/cca9a146-c5a7-493f-849f-94800387b85d)

However, attempting to pay via the saved SEPA payment method that appears under the `Credit card / debit card` radio option will result in the following error on the checkout page:

```
We're not able to process this request. Please refresh the page and try again.
```

And the order notes will include the following error:

```
Error: The PaymentMethod provided (sepa_debit) is not allowed for this PaymentIntent. Please attach a PaymentMethod of one of the following types: card. Alternatively, update the allowed payment_method_types for this PaymentIntent to include "sepa_debit".
```

**Comment by timur27** [comment]
@Lockoloop if everything what Chris mentioned in https://github.com/Automattic/woocommerce-payments/issues/8476#issuecomment-2078046850 is true for your store and saved SEPA token is also showing up under SEPA payment element, then we can use this issue to improve the following:

* make sure saved payment tokens appear only under payment element of the same type
    * saved credit cards appear only under credit card element
    * saved SEPA appear only under SEPA element

**Comment by Lockoloop** [comment]
That seems spot on, great work, guys!

**Comment by Lockoloop** [comment]
Hi @timur27 , has this issue been addressed?

**Comment by csmcneill** [comment]
Howdy @Lockoloop,

This issue has not been addressed yet.

We do not have a planned date for a fix at this time, but I can assure you we will work on it as soon as possible and will notify you when an update is available. 

Additionally, since you opened the issue and commented on it, you should automatically receive [notifications](https://docs.github.com/articles/about-notifications) for future comments and when the issue is moved from `Open` (its current state) to `Closed` (which means the issue has been resolved).

**Comment by timur27** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/8476?planning-poker) with Zenhub @FangedParakeet

**Comment by timur27** [comment]
Hey @Lockoloop! Everything that [Chris mentioned above](https://github.com/Automattic/woocommerce-payments/issues/8476#issuecomment-2179499773) is true. However, we started discussing this issue internally and eventually prepared a fix for it slightly faster than initially planned. It should be included in WooPayments `7.9.0`.

**Comment by Lockoloop** [comment]
Great to hear! Sorry for the previous comment, I was worried the issue had fallen through a crack because you were so fast with fixing the other issue :)
